diff --git a/controls/stig_rhel8.yml b/controls/stig_rhel8.yml
index d7821c2e3b8..fe6b0f01186 100644
--- a/controls/stig_rhel8.yml
+++ b/controls/stig_rhel8.yml
@@ -584,11 +584,6 @@ controls:
         rules:
             - sshd_disable_root_login
         status: automated
-    -   id: RHEL-08-010560
-        levels:
-            - medium
-        title: The auditd service must be running in RHEL 8.
-        status: pending
     -   id: RHEL-08-010561
         levels:
             - medium
diff --git a/products/rhel8/profiles/stig.profile b/products/rhel8/profiles/stig.profile
index 7c89bcbf659..09fa85df181 100644
--- a/products/rhel8/profiles/stig.profile
+++ b/products/rhel8/profiles/stig.profile
@@ -368,9 +368,6 @@ selections:
     # RHEL-08-010550
     - sshd_disable_root_login
 
-    # RHEL-08-010560
-    - service_auditd_enabled
-
     # RHEL-08-010561
     - service_rsyslog_enabled
 
diff --git a/products/rhel9/profiles/stig.profile b/products/rhel9/profiles/stig.profile
index 690991f697b..eb2cac913bd 100644
--- a/products/rhel9/profiles/stig.profile
+++ b/products/rhel9/profiles/stig.profile
@@ -369,9 +369,6 @@ selections:
     # RHEL-08-010550
     - sshd_disable_root_login
 
-    # RHEL-08-010560
-    - service_auditd_enabled
-
     # RHEL-08-010561
     - service_rsyslog_enabled
 
